packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -460,17 +473,21 @@ abstract class VpcBase extends Resource implements IVpc {
       subnets = this.selectSubnetObjectsByType(type);
     }
 
-    if (selection.availabilityZones !== undefined) { // Filter by AZs, if specified
-      subnets = retainByAZ(subnets, selection.availabilityZones);
```

Comment:
Shouldn't this at least lead to the appropriate filter class being prepended to the list?

Strictly speaking this method is `protected`, therefore has a public contract to other people that could have inherited from this class and are relying on its current behavior.

Reply:
Yes you're correct. That functionality is on the line above (474). I did the same for the one-per-az selection option.